cou/steps/__init__.py

```diff
@@ -351,7 +359,7 @@ def __init__(self, description: str):
         :param description: Description of the step.
         :type description: str
         """
-        super().__init__(description=description, parallel=True, coro=None)
+        super().__init__(description=description, parallel=False, coro=None)
```

Comment:
After saying "All sub-steps will run in parallel." seeing `parallel=False` is quite confusing. I think it's important to add a comment here explaining the behavior, as well as improving the docstring of `BaseStep`.